### PR TITLE
makefile(all): add new cases to run_group and check cases in makefile

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: ref #8772

### What is changed and how it works?
1. add new cases to run_group to avoid [serial execution of too many cases at once](https://do.pingcap.net/jenkins/blue/rest/organizations/jenkins/pipelines/pingcap/pipelines/tiflow/pipelines/pull_cdc_integration_test/runs/145/nodes/625/steps/741/log/?start=0)
2. check cases when runnning `make check`
#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -529,6 +529,13 @@ check_third_party_binary_for_engine:
 
 check_engine_integration_test:
 	./engine/test/utils/check_case.sh
+	./engine/test/integration_tests/run_group.sh "check others"
+
+check_dm_integration_test:
+	./dm/tests/run_group.sh "check others"
+
+check_cdc_integration_test:
+	./tests/integration_tests/run_group.sh check "others"
```
Comment (hi-rustin) [inline Makefile:538]: Why don't we unify it to `"check others"` or `check "others"`?

**Review by hi-rustin — COMMENTED** [review]
Thanks!

File: Makefile (hunk 1 above)
Comment (CharlesCheung96) [reply Makefile:538]: Because `./tests/integration_tests/run_group.sh` and `./dm/tests/run_group.sh` have a different number of parameters.

**Review by hi-rustin — APPROVED** [review]
https://github.com/pingcap/tiflow/pull/8794#discussion_r1166535201

Nice to improve it, but not a blocker.

**Review by purelind — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (CharlesCheung96) [reply Makefile:538]: The main reason is that there are ci scripting issues involved here. No matter how you change it, there will always be inconsistent behavior, either inside the script or inside the makefile.

**Review by lance6716 — APPROVED** [review]

**Comment by lance6716** [comment]
for current DM CI failure, please wait https://github.com/pingcap/tidb/pull/43196 and update dependency

**Pull request merged**